### PR TITLE
[WalletConnect issue] - fix for qr code 

### LINKED ISSUE #1787: [autoconnect] - wallet is not correctly connected and QR code remains
**Description**
Wallet is not correctly connected when adding a consecutive wallet via wallet connect. The QR code remains visible eventhough the wallet app shows that the wallet was connected. 
After closing the QR code and trying to connect the wallet again the wallet is connected with the wrong Logo. 
Related to #1680

**How to Reproduce**
1. Go to cowswap
2. Click on connect wallet and choose WC
3. scan QR code, etc..
4. Disconnect wallet
5. Repeat with a new wallet (not only another account but also another wallet provider) 
6. OR code is not closed, close it manually
7. Close modal 
8. try to connect again via WC
9. Wallet is autoconnected but displays the wrong logo #1788 

**Expected behavior**
Connecting a consecutive wallet via WC should be as smooth as adding it for the first time

**Screenshots**
Current issue
https://user-images.githubusercontent.com/34510341/140505618-eb5cf1e9-123d-4340-952b-cd677bc3acfd.mov

1. Connecting with safe
2. Then disconnecting the Safe
3. trying to connect a new wallet(Trust) with WC (0:22-0:40) - The trust app showed that I am connected
4. But QR code remains and cowswap shows that I am not connected
5. Trying to connect again --- auto connected with the Safe which I disconnected before

As comparison in production: 

https://user-images.githubusercontent.com/34510341/140505896-7029e466-8f88-40d7-a9ed-3f615a3a2f49.mov

**Additional context**
See slack conversation 
https://gnosisinc.slack.com/archives/C025G521XQD/p1636037993009000

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Fixes #1787 and probably also #1788

This fixes the problem with random QR code that opens at some point after disconnect from the wallet and messes up the next wallet connection. The main problem seems to be in the previous version of @web3-react/walletconnect-connector so updating the version seems to fix the issue. I've also added removing of the value from local storage for latestProvider on wallet disconnect.

  # To Test

- We can use the steps mentioned in the issue connected to this PR

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @nenadV91 , great!
I do not see WC modal that appears right after conenction when I disconnect a wallet. 
However, I have faced other new issues when testing this PR: 
1. When I change a wallet using 'Change wallet' button in the activity modal, the previously connected  WC wallet becomes disconnected. 
2. Then, when I open connection modal, and then close it, I see an error in the console. And a new QR code is not generated when I open WC connection modal once again
3. I see a lot of yellow logs in the console now. Is this OK?

All these 3 issues you can see in the video: https://watch.screencastify.com/v/8LbMhdQJ1D1XUWFoHCqP

File: package.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
     "@web3-react/fortmatic-connector": "^6.1.6",
     "@web3-react/injected-connector": "^6.0.7",
     "@web3-react/portis-connector": "^6.1.9",
-    "@web3-react/walletconnect-connector": "^6.2.4",
+    "@web3-react/walletconnect-connector": "^6.2.8",
```
Comment (anxolin) [inline package.json:77]: Is this related to the fix? happy to change versions, but not sure if is related

File: src/custom/components/AccountDetails/index.tsx [hunk 2]
```diff
@@ -173,6 +173,7 @@ export default function AccountDetails({
 
   const handleDisconnectClick = () => {
     ;(connector as any).close()
+    localStorage.removeItem(STORAGE_KEY_LAST_PROVIDER)
```
Comment (anxolin) [inline src/custom/components/AccountDetails/index.tsx:176]: Is this the only way users might disconnect? 

should this be in a more central place? (like src/hooks/web3.ts , Web3ReactManager, or sth like that?)

**Comment by anxolin** [comment]
I changed the base branch so this points to v1.5

File: package.json (hunk 1 above)
Comment (nenadV91) [reply package.json:77]: Yes as far as I know the reason for the random QR code was the previous version. Since updating the version that issue didn't occur for me.

File: src/custom/components/AccountDetails/index.tsx (hunk 2 above)
Comment (nenadV91) [reply src/custom/components/AccountDetails/index.tsx:176]: I'm not sure about this but for now I would add this.

**Comment by nenadV91** [comment]
@elena-zh I've added fix it should fix issue 2. I'm not sure I understand the issue 1 or maybe its not there anymore and I would ignore issue 3 for now and fix it later.

**Comment by elena-zh** [comment]
@nenadV91 , steps for the case 1: 
1. connect a wallet using WC
2. open Activity modal 
3. press on the 'Change wallet'' button
4. connect to MM

**Actual result:** WC wallet is disconnected

**Expected result:** WC wallet should be still connected. 
Video of the expected result: https://watch.screencastify.com/v/eYQSurCd5v2yDxDLy21W

**Comment by elena-zh** [comment]
Case 1 looks good to me.

**Comment by nenadV91** [comment]
Because now the localStorage data for walletconnect is removed on changing the wallet button click for some reason. So when you try to connect back from MM to WC there is no data in storage to connect with and it gives you QR code

**Comment by alfetopito** [comment]
@henrypalacios this is the PR Elena mentioned during the call

Feel free to share any xp you might have regarding WC issues from Gnosis Auction if you have any

**Comment by alfetopito** [comment]
> I changed the base branch so this points to v1.5

The branch should be re-based or recreated, otherwise it'll bring stuff from `develop` onto `release/1.5` that we don't want to have yet

**Comment by elena-zh** [comment]
> Because now the localStorage data for walletconnect is removed on changing the wallet button click for some reason. So when you try to connect back from MM to WC there is no data in storage to connect with and it gives you QR code

Then, my question is: why should we need to display 2 different buttons with the same behavior: "Change Wallet' and 'Disconnect'?

@alfetopito , @anxolin , should we leave the implementation as it is or should we keep 'Change Wallet' functionality?

**Comment by nenadV91** [comment]
Recreated branch with PR with the same code for easier merge https://github.com/gnosis/cowswap/pull/1907

**Comment by nenadV91** [comment]
Closing in favor of #1907

**Comment by alfetopito** [comment]
> > Because now the localStorage data for walletconnect is removed on changing the wallet button click for some reason. So when you try to connect back from MM to WC there is no data in storage to connect with and it gives you QR code
> 
> Then, my question is: why should we need to display 2 different buttons with the same behavior: "Change Wallet' and 'Disconnect'?
> 
> @alfetopito , @anxolin , should we leave the implementation as it is or should we keep 'Change Wallet' functionality?

@elena-zh we talked about it and we decided to move forward despite of the new "behaviour"

It came from updating WC version. So, they fixed the bug with double QR codes and introduced this new one.
Nenad is still trying to fix it on our side, but that can come as a hotfix once we have more testing done.

For now, we decided it's best to have the fix included.

**Comment by elena-zh** [comment]
Thank you, @alfetopito , for the explanation. 
I created a separate issue for this case #1909 .

**Comment by anxolin** [comment]
> It came from updating WC version. So, they fixed the bug with double QR codes and introduced this new one.

Thanks for exampling Leandro. I agree with the assessment, although, if the double QR is fixed with the new WC version, why did we also need to delete the last provider in https://github.com/gnosis/cowswap/pull/1907 ? 

was updating WC not enough?

cc/ @nenadV91

**Comment by nenadV91** [comment]
Anxo I think there is a misunderstanding, the new issue with the wallet change is not caused by deleting of lastProvider. Instead with the update of @web3-react/walletconnect-connector package now property called `walletconnect` which is stored in localStorage by some of the third party libraries (probably this one) is now automatically deleted when you change wallet and this is not done by us, meaning we didn't change any code related to this new issue.
